tests/app/database/test_threads.py

```diff
@@ -11,41 +11,55 @@
 from neuroagent.app.routers.database.schemas import GetThreadsOutput
 
 
-def test_create_thread(patch_required_env, app_client, db_connection):
+def test_create_thread(patch_required_env, httpx_mock, app_client, db_connection):
     test_settings = Settings(
         db={"prefix": db_connection},
     )
     app.dependency_overrides[get_settings] = lambda: test_settings
+    httpx_mock.add_response(
+        url=f"{test_settings.virtual_lab.get_project_url}/test_vlab/projects/test_project"
+    )
     with app_client as app_client:
         # Create a thread
-        create_output = app_client.post("/threads/").json()
+        create_output = app_client.post(
+            "/threads/?virtual_lab_id=test_vlab&project_id=test_project"
+        ).json()
     assert create_output["thread_id"]
     assert create_output["user_sub"] == "dev"
     assert create_output["title"] == "title"
     assert create_output["timestamp"]
 
 
-def test_get_threads(patch_required_env, app_client, db_connection):
+@pytest.mark.httpx_mock(can_send_already_matched_responses=True)
```

Comment:
What does this decorator do ?

Reply:
There will be two validation calls for the two created threads. The new `httpx_mock` version only allows for a mock to be used once. 
(I tested it is needed here.)

Reply:
Ah so basically it mocks it "forever" (within the test) and not just once ? Useful indeed.